src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -103,6 +103,90 @@ static void handle_signal(int signum)
   }
 }
 
+#ifdef CEPH_HAVE_SPLICE
+static bool tcp_pipe(int *_source_fd, int *_sink_fd) {
+  if (_source_fd == nullptr || _sink_fd == nullptr) {
+    return false;
+  }
+  int source_fd = -1;
+  int sink_fd = -1;
+  int server_fd = -1;
+  struct sockaddr_in server_addr;
+  memset(&server_addr, 0, sizeof(struct sockaddr_in));
+  server_addr.sin_family = AF_INET;
+  server_addr.sin_addr.s_addr = INADDR_ANY;
+  server_addr.sin_port = 0;
+  try {
+    server_fd = socket(AF_INET, SOCK_STREAM, 0);
+    if (server_fd < 0) {
+      throw (__LINE__);
+    }
+    int enable = 1;
+    if (setsockopt(server_fd, SOL_SOCKET, SO_REUSEADDR, &enable, sizeof(int)) != 0) {
+      throw __LINE__;
+    }
+    if (bind(server_fd, (const struct sockaddr*) &server_addr, sizeof(server_addr)) != 0) {
+      throw __LINE__;
+    }
+    socklen_t len;
+    if (getsockname(server_fd, (struct sockaddr*)&server_addr, &len) != 0) {
+      throw __LINE__;
+    }
+    if(server_addr.sin_port == 0) {
+      derr << "Port == 0" << dendl;
+    }
+    if (listen(server_fd, 1) != 0) {
```

Comment:
@amitkumar50 Which is good, since it only connects once (line 148) and closes (line 161)